### PR TITLE
Fix security vulnerabilities

### PR DESCRIPTION
**Pull request by majetideepak**
IBM code scan found a few issues.

- Use of uninitialized variables: It is always safe to initialize variables. Most instances fixed in the PR get initialized
   in a function but depending on a function is unsafe.
- Detect and handle memory allocation errors
- Bitwise operations (~, <<, >>, &, ^ and |) and their combinations with assign operator (<<=, >>=, &=, ^= and |=) are not normally meaningful on signed integers. Bitwise operators should be used only with unsigned integer operands, as the result of bitwise operations on signed integers are implementation-defined.

### CODE REVIEW

**Comment by kgpai** [comment]
@majetideepak Is it possible to add what cve's this fixes ?

**Comment by majetideepak** [comment]
@kgpai I updated the description. Thanks!

File: velox/dwio/dwrf/reader/SelectiveStringDirectColumnReader.cpp [hunk 1]
```diff
@@ -237,8 +237,8 @@ inline bool SelectiveStringDirectColumnReader::try8Consecutive(
   const char* data = bufferStart_ + start + bytesToSkip_;
   if constexpr (!sparse) {
     auto* lengths = rawLengths_ + rows[row];
-    uint16_t offsets[9];
-    bool gt4;
+    uint16_t offsets[9] = {0};
```
Comment (Yuhta) [inline velox/dwio/dwrf/reader/SelectiveStringDirectColumnReader.cpp:240]: These are left uninitialized for performance reason.  Can we suppress the warning for low level code like this?

Comment (majetideepak) [reply velox/dwio/dwrf/reader/SelectiveStringDirectColumnReader.cpp:240]: Yes, we can suppress warnings for low-level code like this.  I will remove this change.

**Review by czentgr — APPROVED** [review]

**Review by kgpai — APPROVED** [review]
LGTM.

**Pull request closed without merging**